src/qz/printer/status/StatusMonitor.java

```diff
@@ -74,45 +78,25 @@ public synchronized static void closeNotificationThreads() {
         }
     }
 
-    public synchronized static boolean startListening(SocketConnection connection, JSONObject params) throws JSONException {
+    public synchronized static boolean isListening(SocketConnection connection) {
+        return statusSessions.containsKey(connection);
+    }
+
+    public synchronized static boolean startListening(SocketConnection connection, Session session, JSONObject params) throws JSONException {
+        statusSessions.putIfAbsent(connection, new StatusSession(session));
         JSONArray printerNames = params.getJSONArray("printerNames");
```

Comment:
We should do all JSON work BEFORE editing any list, just in case we throw a JSONException